src/System.Private.CoreLib/shared/System/DefaultBinder.cs

```diff
@@ -1204,6 +1204,67 @@ private static bool CreateParamOrder(int[] paramOrder, ParameterInfo[] pars, str
             return true;
         }
 
+        // CanChangePrimitive
+        // This will determine if the source can be converted to the target type
+        internal static bool CanChangePrimitive(Type source, Type target)
+        {
+            if ((source == typeof(IntPtr) && target == typeof(IntPtr)) ||
+                (source == typeof(UIntPtr) && target == typeof(UIntPtr)))
+                return true;
+
+            Primitives widerCodes = s_primitiveConversions[(int)(Type.GetTypeCode(source))];
```

Comment:
This new implementation will have different behavior for enums. The C++ code in `DBCanConvertPrimitive` called `GetSignatureCorElementType` that returns VALUETYPE for enums, but `GetTypeCode` will return the underlying type.

E.g. `CanChangePrimitive(typeof(MyEnum), typeof(MyEnum))` returns false today, but it is going to return true with this change.

I think that this should be ok because of none of the callers depend on this, but it would be good if you can double check it too.

Reply:
@jkotas everywhere where this method is used the second argument (Type) is checked for `IsPrimitive` (which is `false` for `Enum`) so this case is not possible.